src/main/java/com/code_intelligence/jazzer/mutation/mutator/collection/MapMutatorFactory.java

```diff
@@ -0,0 +1,238 @@
+        });
+  }
+
+  private static final class MapMutator<K, V> extends SerializingInPlaceMutator<Map<K, V>> {
+    private static final int DEFAULT_MIN_SIZE = 0;
+    private static final int DEFAULT_MAX_SIZE = 1000;
+
+    private final SerializingMutator<K> keyMutator;
+    private final SerializingMutator<V> valueMutator;
+    private final int minSize;
+    private final int maxSize;
+
+    MapMutator(SerializingMutator<K> keyMutator, SerializingMutator<V> valueMutator, int minSize,
+        int maxSize) {
+      this.keyMutator = keyMutator;
+      this.valueMutator = valueMutator;
+      this.minSize = Math.max(minSize, DEFAULT_MIN_SIZE);
+      this.maxSize = Math.min(maxSize, DEFAULT_MAX_SIZE);
+
+      require(maxSize >= 1, "Max. size needs to be greater than 0");
+      require(minSize >= 0, "Min. size needs to be greater or equal 0");
+      require(minSize <= maxSize,
+          format("Min. size %d needs to be smaller or equal than max. %d", minSize, maxSize));
+    }
+
+    @Override
+    public Map<K, V> read(DataInputStream in) throws IOException {
+      int size = Math.min(Math.max(in.readInt(), minSize), maxSize);
+      Map<K, V> map = new HashMap<>(size);
+      for (int i = 0; i < size; i++) {
+        map.put(keyMutator.read(in), valueMutator.read(in));
+      }
+      // Wrap in an immutable view for additional protection against accidental mutation in fuzz
+      // tests.
+      return toImmutableMapView(map);
+    }
+
+    @Override
+    public void write(Map<K, V> map, DataOutputStream out) throws IOException {
+      out.writeInt(map.size());
+      for (Map.Entry<K, V> entry : map.entrySet()) {
+        keyMutator.write(entry.getKey(), out);
+        valueMutator.write(entry.getValue(), out);
+      }
+    }
+
+    @Override
+    protected Map<K, V> makeDefaultInstance() {
+      // Wrap in an immutable view for additional protection against accidental mutation in fuzz
+      // tests.
+      return toImmutableMapView(new HashMap<>(maxInitialSize()));
+    }
+
+    @Override
+    public void initInPlace(Map<K, V> reference, PseudoRandom prng) {
+      int targetSize = prng.closedRange(minInitialSize(), maxInitialSize());
+      Map<K, V> map = underlyingMutableList(reference);
+      map.clear();
+      for (int i = 0; i < targetSize; i++) {
+        initElement(map, prng);
+      }
+    }
+
+    @Override
+    public void mutateInPlace(Map<K, V> reference, PseudoRandom prng) {
+      Map<K, V> map = underlyingMutableList(reference);
+      if (map.isEmpty()) {
+        initElement(map, prng);
+      } else if (!prng.trueInOneOutOf(4)) {
+        // Choose a random entry, and then key or value, to mutate.
+        int i = prng.indexIn(map.size());
+        map.entrySet().stream().skip(i).findFirst().ifPresent(
+            entry -> mutateElement(entry, map, prng));
+      } else {
+        // This will increase the map size up to max and then oscillate around it.
```

Comment:
This oscillation behavior is a bit weird. I know we have it for the list mutator, but we probably want to get rid of it sooner rather than later. Could you turn this into a FIXME and create a follow-up ticket?

Reply:
Done